.github/scripts/filter_test_configs.py

```diff
@@ -191,12 +200,86 @@ def set_periodic_modes(
     return scheduled_test_matrix
 
 
+def mark_unstable_jobs(
+    workflow: str, job_name: str, test_matrix: Dict[str, List[Any]]
+) -> Dict[str, List[Any]]:
+    """
+    Check the list of unstable jobs and mark them accordingly. Note that if a job
+    is unstable, all its dependents will also be marked accordingly
+    """
+    return process_jobs(
+        workflow=workflow,
+        job_name=job_name,
+        test_matrix=test_matrix,
+        issue_type=IssueType.UNSTABLE,
+        url=UNSTABLE_JOBS_URL,
+    )
+
+
 def remove_disabled_jobs(
     workflow: str, job_name: str, test_matrix: Dict[str, List[Any]]
 ) -> Dict[str, List[Any]]:
     """
     Check the list of disabled jobs, remove the current job and all its dependents
-    if it exists in the list. The list of disabled jobs is as follows:
+    if it exists in the list
+    """
+    return process_jobs(
+        workflow=workflow,
+        job_name=job_name,
+        test_matrix=test_matrix,
+        issue_type=IssueType.DISABLED,
+        url=DISABLED_JOBS_URL,
+    )
+
+
+def _filter_jobs(
+    test_matrix: Dict[str, List[Any]],
+    issue_type: IssueType,
+    target_cfg: Optional[str] = None,
+) -> Dict[str, List[Any]]:
+    """
+    An utility function used to actually apply the job filter
+    """
+    # The result will be stored here
+    filtered_test_matrix: Dict[str, List[Any]] = {"include": []}
+
+    # This is an issue to disable a CI job
+    if issue_type == IssueType.DISABLED:
+        # If there is a target config, disable (remove) only that
+        if target_cfg:
+            # Remove the target config from the test matrix
+            filtered_test_matrix["include"] = [
+                r for r in test_matrix["include"] if r.get("config", "") != target_cfg
+            ]
+
+        return filtered_test_matrix
+
+    if issue_type == IssueType.UNSTABLE:
+        for r in test_matrix["include"]:
+            cpy = r.copy()
+
+            if (target_cfg and r.get("config", "") == target_cfg) or not target_cfg:
+                # If there is a target config, only mark that as unstable, otherwise,
+                # mark everything as unstable
+                cpy[IssueType.UNSTABLE.value] = IssueType.UNSTABLE.value
```

Comment:
A dict with `{"unstable": "unstable"}` seems redundant.

What about `{"issue_type": "unstable"}` (which lets you show the "disabled" status and other potential future statuses

Reply:
Ah yeah, now that I think about it `{"unstable": "unstable"}` sounds weird.  I copy this convention from ``{"mem_leak_check": "mem_leak_check"}`.  Let me see if `{"issue_type": "unstable"}` works (it should)

Reply:
On second thought, I think this is a minor thing quirk that we can keep as it makes is slightly easier to check if the job has a status, i.e. `"mem_leak_check" in  test_matrix["include"]` or `"unstable" in  test_matrix["include"]`.